storage/client.go

```diff
@@ -308,6 +308,15 @@ func (c Client) buildCanonicalizedResourceTable(uri string) (string, error) {
 		cr += u.EscapedPath()
 	}
 
+	params, err := url.ParseQuery(u.RawQuery)
+
+	// search for "comp" parameter, if exists then add it to canonicalizedresource
+	for key := range params {
```

Comment:
Could you also add a test for this inside `Test_buildCanonicalizedResourceTable` in `client_test.go`, please?